tensordict/nn/storage.py

```diff
@@ -0,0 +1,346 @@
+        for index, value in zip(indices.tolist(), values.unbind(0)):
+            self.tensor_dict[index] = value
+
+    def __len__(self) -> None:
+        return len(self.tensor_dict)
+
+    def contain(self, indices: torch.Tensor) -> torch.Tensor:
+        res: List[bool] = []
+        for index in indices.tolist():
+            res.append(index in self.tensor_dict)
+
+        return torch.tensor(res, dtype=torch.int64)
+
+
+class FixedStorage(nn.Module, TensorStorage[torch.Tensor, torch.Tensor]):
+    """A Fixed Tensor Storage.
+
+    This is storage that backed by nn.Embedding and hence can be in any device that
+    nn.Embedding supports. The size of storage is fixed and cannot be extended.
+    """
+
+    def __init__(
+        self, embedding: nn.Embedding, init_fm: Callable[[torch.Tensor], torch.Tensor]
+    ):
+        super().__init__()
+        self.embedding = embedding
+        self.num_embedding = embedding.num_embeddings
+        self.flag = None
+        self.init_fm = init_fm
+        self.clear()
+
+    def clear(self):
+        self.init_fm(self.embedding.weight)
+        self.flag = torch.zeros((self.embedding.num_embeddings, 1), dtype=torch.int64)
+
+    def to_index(self, item: torch.Tensor) -> torch.Tensor:
+        return torch.remainder(item.to(torch.int64), self.num_embedding).to(torch.int64)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.embedding(self.to_index(x))
+
+    def __getitem__(self, item: torch.Tensor) -> torch.Tensor:
+        return self.forward(item)
+
+    def __setitem__(self, item: torch.Tensor, value: torch.Tensor) -> None:
+        if value.shape[-1] != self.embedding.embedding_dim:
+            raise ValueError(
+                "The shape value does not match with storage cell shape, "
+                f"expected {self.embedding.embedding_dim} but got {value.shape[-1]}!"
+            )
+        index = self.to_index(item)
+        with torch.no_grad():
+            self.embedding.weight[index, :] = value
+            self.flag[index] = 1
+
+    def __len__(self) -> int:
+        return torch.sum(self.flag).item()
+
+    def contain(self, item: torch.Tensor) -> torch.Tensor:
+        index = self.to_index(item)
+        return self.flag[index]
+
+
+class BinaryToDecimal(torch.nn.Module):
+    """A Module to convert binaries encoded tensors to decimals.
+
+    This is a utility class that allow to convert a binary encoding tensor (e.g. `1001`) to
+    its decimal value (e.g. `9`)
+    """
+
+    def __init__(
+        self,
+        num_bits: int,
+        device: torch.device,
+        dtype: torch.dtype,
+        convert_to_binary: bool,
+    ):
+        super().__init__()
+        self.convert_to_binary = convert_to_binary
+        self.bases = 2 ** torch.arange(num_bits - 1, -1, -1, device=device, dtype=dtype)
+        self.num_bits = num_bits
+        self.zero_tensor = torch.zeros((1,))
```

Comment:
shouldn't we put this on device? What's the purpose of this tensor?

Reply:
yes, we should, will fix it.

We could instantiate it in L158, but I thought instantiate it in init would improve efficiency. 

Let me know your thoughts and I can move it there.